subsys/net/lib/lwm2m/lwm2m_registry.c

```diff
@@ -1156,8 +1156,8 @@ static int lwm2m_engine_get(const struct lwm2m_obj_path *path, void *buf, uint16
 			break;
 
 		case LWM2M_RES_TYPE_STRING:
-			strncpy(buf, data_ptr, buflen - 1);
-			((char *)buf)[buflen - 1] = '\0';
+			strncpy(buf, data_ptr, data_len - 1);
+			((char *)buf)[data_len - 1] = '\0';
 			break;
```

Comment:
Is this an intended change, where is ensured that data_len <= buflen?

Reply:
There's a generic check `lwm2m_check_buf_sizes()` above.

Reply:
@ssharks Yes, this is intentional.

buflen refers to a buffer size where we are going to write. `data_len` is the amount of data we have in registry.
If binary write to string buffer has been made, I still need to ensure that my `strncpy()` does not write more than what we have stored.

So if we write 10 bytes of garbage without terminating character, then ask for read of 20 bytes, previously we got 20 bytes of garbage. So the limiting size value should be the `data_len` variable. Not `buflen`.

Buffer sizes are already verified in `lwm2m_check_buf_sizes()`.